Source/Blazorise/Fluent/FluentObjectFit.cs

```diff
@@ -0,0 +1,265 @@
+    /// <returns>List of classnames for the given rules and the class provider.</returns>
+    string Class( IClassProvider classProvider );
+}
+
+/// <summary>
+/// Breakpoints allowed for object-fit.
+/// </summary>
+public interface IFluentObjectFitOnBreakpoint :
+    IFluentObjectFit
+{
+    /// <summary>
+    /// Valid on all devices. (extra small)
+    /// </summary>
+    IFluentObjectFitWithSize OnMobile { get; }
+
+    /// <summary>
+    /// Breakpoint on tablets (small).
+    /// </summary>
+    IFluentObjectFitWithSize OnTablet { get; }
+
+    /// <summary>
+    ///  Breakpoint on desktop (medium).
+    /// </summary>
+    IFluentObjectFitWithSize OnDesktop { get; }
+
+    /// <summary>
+    /// Breakpoint on widescreen (large).
+    /// </summary>
+    IFluentObjectFitWithSize OnWidescreen { get; }
+
+    /// <summary>
+    /// Breakpoint on large desktops (extra large).
+    /// </summary>
+    IFluentObjectFitWithSize OnFullHD { get; }
+
+    /// <summary>
+    /// Breakpoint on extra large desktops (extra extra large).
+    /// </summary>
+    IFluentObjectFitWithSize OnQuadHD { get; }
+}
+
+/// <summary>
+/// Sizes allowed for object-fit.
+/// </summary>
+public interface IFluentObjectFitWithSize
+    : IFluentObjectFit
+{
+    /// <summary>
+    /// The default object fit behavior.
+    /// </summary>
+    IFluentObjectFitOnBreakpoint Default { get; }
+
+    /// <summary>
+    /// The object does not scale to fit the container and may be clipped.
+    /// </summary>
+    IFluentObjectFitOnBreakpoint None { get; }
+
+    /// <summary>
+    /// The object is scaled to maintain its aspect ratio while fitting within the container.
+    /// </summary>
+    IFluentObjectFitOnBreakpoint Contain { get; }
+
+    /// <summary>
+    /// The object is scaled to cover the entire container, possibly clipping some parts.
+    /// </summary>
+    IFluentObjectFitOnBreakpoint Cover { get; }
+
+    /// <summary>
+    /// The object is scaled to completely fill the container. 
+    /// Its aspect ratio is not preserved, and the object may be stretched or compressed.
+    /// </summary>
+    IFluentObjectFitOnBreakpoint Fill { get; }
+
+    /// <summary>
+    /// The object is scaled to fill the container without preserving its aspect ratio.
+    /// </summary>
+    IFluentObjectFitOnBreakpoint Scale { get; }
+}
+
+/// <summary>
+/// Defines the size information for the breakpoint.
+/// </summary>
+public sealed class ObjectFitDefinition
+{
+    /// <summary>
+    /// Defines the breakpoint on which the object-fit will apply.
+    /// </summary>
+    public Breakpoint Breakpoint { get; set; }
+}
+
+/// <summary>
+/// Default implementation of fluent object-fit builder.
+/// </summary>
+public class FluentObjectFit :
+    IFluentObjectFitOnBreakpoint,
+    IFluentObjectFitWithSize
+{
+    #region Members
+
+    private ObjectFitDefinition currentObjectFitDefintion;
+
+    private readonly Dictionary<ObjectFitType, ObjectFitDefinition> rules = new();
```

Comment:
Just making sure. If an instance of this exists, then it makes sense to always have the Dictionary newed up? As typically it will always at least have a record added, is this right?

Reply:
Blazor will always create a new instance of the object. That is something that we still haven't fixed. But, since it doesn't affect performance too much it is not that big of a deal. So I never really bother.

In the long run, we will definitely have to find a way to keep and cache fluent instances between each render.